### PR TITLE
Add Azure Files tutorial

### LINKED ISSUE #126: Document how to use Enclave to access Azure Files
Parent issue:

- https://github.com/enclave-networks/customer-support/issues/124

Describe the options, Azure Files Private Endpoint vs. IP whitelisted access. Explain the pros and cons of each approach and make a firm recommendation for our customers.

### PR DESCRIPTION
**Pull request by enclave-alistair**
Closes #126

### CODE REVIEW

File: docs/tutorials/secure-private-access-to-azure-files.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+- A single subnet in that network of `10.1.0.0/24`.
+- An Azure Files instance with a DNS name of `mystorage.file.core.windows.net`
+  
+## Set up a Private Endpoint
+
+The first step in deploying this solution is to set up a Private Endpoint in your virtual network, that points to your Azure Files instance. There is already good information on doing this in the [Azure documentation](https://learn.microsoft.com/en-us/azure/storage/files/storage-files-networking-endpoints?tabs=azure-portal#create-a-private-endpoint), but I'll outline the precise steps here.
+
+!!! tip
+
+    The only significant deviation of these steps from the default instructions is to set a **Static IP** for the private endpoint for you to reference later.
```
Comment (enclave-charlie) [inline docs/tutorials/secure-private-access-to-azure-files.md:35]: I'd re-link the Azure documentation in "default instructions", or specify "The only significant deviation of these steps from the documentation provided by Azure". Doesn't feel clear enough to me what the "default instructions" are.

File: docs/tutorials/secure-private-access-to-azure-files.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+     ![Azure Files Private Endpoint Basic Settings](../images/guides/azure-files/private-endpoint-basic.png){: .screenshot}
+
+3. On the Resource tab, select the **file** sub-resource.
+
+     ![Azure Files Private Endpoint Resource Settings](../images/guides/azure-files/private-endpoint-resource.png){: .screenshot}
+
+4. Under Virtual Network, choose which network the private endpoint should be accessible from. This must be the same the network in which your Enclave Gateway will reside.
+
+     Select **Statically allocate IP address** and choose an IP address for the private endpoint. Following the diagram above, we are allocating **10.1.0.100**, within our 10.1.0.0/24 subnet.
```
Comment (enclave-charlie) [inline docs/tutorials/secure-private-access-to-azure-files.md:51]: "Following the diagram above, allocate **10.1.0.100**, within our 10.1.0.0/24 subnet." feels a bit unclear to me.

"For the purposes of this tutorial, we've allocated **10.1.0.100**, within our 10.1.0.0/24 subnet."? 
"As per the diagram above, we've allocated **10.1.0.100**, within our 10.1.0.0/24 subnet."?

File: docs/tutorials/secure-private-access-to-azure-files.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+## Define your Policy
+
+Once your gateway and private endpoint are set up, you can proceed to define your Enclave policy.
+
+1. Define a **Gateway Access Policy** from a ==users== tag to the gateway you've deployed.
+
+    ![Adding the users tag and the deployed gateway to the policy](../images/guides/azure-files/policy-members.png){: .screenshot}
+   
+2. Configure a subnet filter on the policy to only allow access to your private endpoint (**10.1.0.100**) within your **10.1.0.0/24** subnet:
```
Comment (enclave-charlie) [inline docs/tutorials/secure-private-access-to-azure-files.md:88]: Should **10.1.0.0/24** be in brackets too?

File: docs/tutorials/secure-private-access-to-azure-files.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+3. Press save to create the policy.
+
+!!! note
+
+    We've found that the performance of Windows explorer when applying *just* the SMB port of TCP 445 is not great, that's why we've left the allowed traffic to the private endpoint as **All Traffic**. This isn't a problem in this instance, because Azure Files is a managed service, and so the attack surface is minimal.
+
+## Remove Public Access to your Storage Account
+
+Finally, you can remove public access to your storage account, preventing any access to your file shares from the internet, ensuring that all access is mediated by Enclave.
```
Comment (enclave-charlie) [inline docs/tutorials/secure-private-access-to-azure-files.md:100]: "and ensuring" flows a bit better to my ear:

"Finally, you can remove public access to your storage account, preventing any access to your file shares from the internet and ensuring that all access is mediated by Enclave."

**Review by enclave-charlie — CHANGES_REQUESTED** [review]

File: docs/tutorials/secure-private-access-to-azure-files.md (hunk 3 above)
Comment (enclave-alistair) [reply docs/tutorials/secure-private-access-to-azure-files.md:88]: Replaced the brackets with a comma block? Brackets were kind of a callback I guess, but the subnet shouldn't have them.

Comment (enclave-charlie) [reply docs/tutorials/secure-private-access-to-azure-files.md]: Why shouldn't the subnet have them? I thought it was also an example chosen for the tutorial?

**Review by enclave-charlie — APPROVED** [review]

**Pull request merged**